In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("../Archivos/union_datos.csv")

---

### Analizando nulos columnas

---

In [3]:
#Definimos función para ver nulos y otra renombrar columnas
def nulos(col_ini, col_fin):
    no_nulos = df.loc[:, col_ini : col_fin].notnull().sum()/df.shape[0]*100 
    for col in no_nulos.index:
        print (col, '--->', df[col].value_counts().index[0], '--->', (df.loc[:,col].notnull().sum()/df.shape[0]*100).round(2),'%' )

def renombrar(col_ini, col_fin):
        lista =[]
        no_nulos = df.loc[:, col_ini : col_fin].notnull().sum()/df.shape[0]*100    
        for col in no_nulos.index:
            nombre = df[col].value_counts().index[0]
            lista.append(re.findall('Q\d{2}',col)[0]+'_'+nombre.lower().strip().replace(' ','_'))
        columnas = dict(zip(no_nulos.index, lista))
        return columnas

### Q39 donde compartes publicamente tu trabajo

---

In [4]:
nulos('Q39_Part_1','Q39_OTHER')

Q39_Part_1 --->  Plotly Dash  ---> 1.13 %
Q39_Part_2 --->  Streamlit  ---> 1.49 %
Q39_Part_3 --->  NBViewer  ---> 0.52 %
Q39_Part_4 --->  GitHub  ---> 17.66 %
Q39_Part_5 --->  Personal blog  ---> 2.71 %
Q39_Part_6 --->  Kaggle  ---> 11.8 %
Q39_Part_7 --->  Colab  ---> 7.12 %
Q39_Part_8 --->  Shiny  ---> 1.17 %
Q39_Part_9 ---> I do not share my work publicly ---> 12.19 %
Q39_OTHER ---> Other ---> 1.18 %


### Q39 donde compartes publicamente tu trabajo
    Q39_Part_1 --->  Plotly Dash  ---> 1.13 %
    Q39_Part_2 --->  Streamlit  ---> 1.49 %
    Q39_Part_3 --->  NBViewer  ---> 0.52 %
    Q39_Part_4 --->  GitHub  ---> 17.66 %
    Q39_Part_5 --->  Personal blog  ---> 2.71 %
    Q39_Part_6 --->  Kaggle  ---> 11.8 %
    Q39_Part_7 --->  Colab  ---> 7.12 %
    Q39_Part_8 --->  Shiny  ---> 1.17 %
    Q39_Part_9 ---> I do not share my work publicly ---> 12.19 %
    Q39_OTHER ---> Other ---> 1.18 %


### Q40 en que plataformas has realizado cursos de data science

---

In [5]:
nulos('Q40_Part_1','Q40_OTHER')

Q40_Part_1 ---> Coursera ---> 37.15 %
Q40_Part_2 ---> edX ---> 11.66 %
Q40_Part_3 ---> Kaggle Learn Courses ---> 33.38 %
Q40_Part_4 ---> DataCamp ---> 15.39 %
Q40_Part_5 ---> Fast.ai ---> 4.45 %
Q40_Part_6 ---> Udacity ---> 10.79 %
Q40_Part_7 ---> Udemy ---> 24.79 %
Q40_Part_8 ---> LinkedIn Learning ---> 10.37 %
Q40_Part_9 ---> Cloud-certification programs (direct from AWS, Azure, GCP, or similar) ---> 6.95 %
Q40_Part_10 ---> University Courses (resulting in a university degree) ---> 18.8 %
Q40_Part_11 ---> None ---> 6.78 %
Q40_OTHER ---> Other ---> 8.01 %


### Q40 en que plataformas has realizado cursos de data science
    Q40_Part_1 ---> Coursera ---> 37.15 %
    Q40_Part_2 ---> edX ---> 11.66 %
    Q40_Part_3 ---> Kaggle Learn Courses ---> 33.38 %
    Q40_Part_4 ---> DataCamp ---> 15.39 %
    Q40_Part_5 ---> Fast.ai ---> 4.45 %
    Q40_Part_6 ---> Udacity ---> 10.79 %
    Q40_Part_7 ---> Udemy ---> 24.79 %
    Q40_Part_8 ---> LinkedIn Learning ---> 10.37 %
    Q40_Part_9 ---> Cloud-certification programs (direct from AWS, Azure, GCP, or similar) ---> 6.95 %
    Q40_Part_10 ---> University Courses (resulting in a university degree) ---> 18.8 %
    Q40_Part_11 ---> None ---> 6.78 %
    Q40_OTHER ---> Other ---> 8.01 %


### Q27B - En que plataformas de cloud computing esperas familiarizarte en los próximos 2 años

---

In [6]:
nulos('Q27_B_Part_1','Q27_B_OTHER')

Q27_B_Part_1 --->  Amazon Web Services (AWS)  ---> 28.85 %
Q27_B_Part_2 --->  Microsoft Azure  ---> 21.75 %
Q27_B_Part_3 --->  Google Cloud Platform (GCP)  ---> 28.81 %
Q27_B_Part_4 --->  IBM Cloud / Red Hat  ---> 9.97 %
Q27_B_Part_5 --->  Oracle Cloud  ---> 6.59 %
Q27_B_Part_6 --->  SAP Cloud  ---> 3.5 %
Q27_B_Part_7 --->  VMware Cloud  ---> 3.02 %
Q27_B_Part_8 --->  Salesforce Cloud  ---> 3.05 %
Q27_B_Part_9 --->  Alibaba Cloud  ---> 2.43 %
Q27_B_Part_10 --->  Tencent Cloud  ---> 2.12 %
Q27_B_Part_11 ---> None ---> 8.62 %
Q27_B_OTHER ---> Other ---> 1.04 %


### Q27B - En que plataformas de cloud computing esperas familiarizarte en los próximos 2 años
    Q27_B_Part_1 --->  Amazon Web Services (AWS)  ---> 28.85 %
    Q27_B_Part_2 --->  Microsoft Azure  ---> 21.75 %
    Q27_B_Part_3 --->  Google Cloud Platform (GCP)  ---> 28.81 %
    Q27_B_Part_4 --->  IBM Cloud / Red Hat  ---> 9.97 %
    Q27_B_Part_5 --->  Oracle Cloud  ---> 6.59 %
    Q27_B_Part_6 --->  SAP Cloud  ---> 3.5 %
    Q27_B_Part_7 --->  VMware Cloud  ---> 3.02 %
    Q27_B_Part_8 --->  Salesforce Cloud  ---> 3.05 %
    Q27_B_Part_9 --->  Alibaba Cloud  ---> 2.43 %
    Q27_B_Part_10 --->  Tencent Cloud  ---> 2.12 %
    Q27_B_Part_11 ---> None ---> 8.62 %
    Q27_B_OTHER ---> Other ---> 1.04 %


### Q29B - En que productos específicos de cloud computing esperas familiarizarte en los próximos 2 años

---

In [7]:
nulos('Q29_B_Part_1','Q29_B_OTHER')

Q29_B_Part_1 --->  Amazon Elastic Compute Cloud (EC2)  ---> 20.71 %
Q29_B_Part_2 --->  Microsoft Azure Virtual Machines  ---> 22.3 %
Q29_B_Part_3 --->  Google Cloud Compute Engine  ---> 28.86 %
Q29_B_Part_4 ---> None ---> 3.73 %
Q29_B_OTHER ---> Other ---> 0.3 %


### Q29B - En que productos específicos de cloud computing esperas familiarizarte en los próximos 2 años
    Q29_B_Part_1 --->  Amazon Elastic Compute Cloud (EC2)  ---> 20.71 %
    Q29_B_Part_2 --->  Microsoft Azure Virtual Machines  ---> 22.3 %
    Q29_B_Part_3 --->  Google Cloud Compute Engine  ---> 28.86 %
    Q29_B_Part_4 ---> None ---> 3.73 %
    Q29_B_OTHER ---> Other ---> 0.3 %

### Q31B - En que productos machine learning esperas familiarizarte en los próximos 2 años

---

In [9]:
nulos('Q31_B_Part_1','Q31_B_OTHER')

Q31_B_Part_1 --->  Amazon SageMaker  ---> 14.05 %
Q31_B_Part_2 --->  Azure Machine Learning Studio  ---> 19.36 %
Q31_B_Part_3 --->  Google Cloud Vertex AI ---> 20.42 %
Q31_B_Part_4 --->  DataRobot ---> 7.05 %
Q31_B_Part_5 --->  Databricks ---> 8.48 %
Q31_B_Part_6 --->  Dataiku ---> 2.94 %
Q31_B_Part_7 --->  Alteryx ---> 3.05 %
Q31_B_Part_8 --->  Rapidminer ---> 3.92 %
Q31_B_Part_9 ---> None ---> 15.83 %
Q31_B_OTHER ---> Other ---> 1.34 %


### Q31B - En que productos machine learning esperas familiarizarte en los próximos 2 años
    Q31_B_Part_1 --->  Amazon SageMaker  ---> 14.05 %
    Q31_B_Part_2 --->  Azure Machine Learning Studio  ---> 19.36 %
    Q31_B_Part_3 --->  Google Cloud Vertex AI ---> 20.42 %
    Q31_B_Part_4 --->  DataRobot ---> 7.05 %
    Q31_B_Part_5 --->  Databricks ---> 8.48 %
    Q31_B_Part_6 --->  Dataiku ---> 2.94 %
    Q31_B_Part_7 --->  Alteryx ---> 3.05 %
    Q31_B_Part_8 --->  Rapidminer ---> 3.92 %
    Q31_B_Part_9 ---> None ---> 15.83 %
    Q31_B_OTHER ---> Other ---> 1.34 %

### Q32B - En que productos de big data esperas familiarizarte en los próximos 2 años

---

In [10]:
nulos('Q32_B_Part_1','Q32_B_OTHER')

Q32_B_Part_1 ---> MySQL  ---> 23.62 %
Q32_B_Part_2 ---> PostgreSQL  ---> 12.66 %
Q32_B_Part_3 ---> SQLite  ---> 11.69 %
Q32_B_Part_4 ---> Oracle Database  ---> 8.87 %
Q32_B_Part_5 ---> MongoDB  ---> 16.51 %
Q32_B_Part_6 ---> Snowflake  ---> 4.18 %
Q32_B_Part_7 ---> IBM Db2  ---> 4.6 %
Q32_B_Part_8 ---> Microsoft SQL Server  ---> 12.1 %
Q32_B_Part_9 ---> Microsoft Azure SQL Database  ---> 10.32 %
Q32_B_Part_10 ---> Microsoft Azure Cosmos DB  ---> 5.97 %
Q32_B_Part_11 ---> Amazon Redshift  ---> 4.66 %
Q32_B_Part_12 ---> Amazon Aurora  ---> 4.55 %
Q32_B_Part_13 ---> Amazon DynamoDB  ---> 5.75 %
Q32_B_Part_14 ---> Amazon RDS  ---> 8.64 %
Q32_B_Part_15 ---> Google Cloud BigQuery  ---> 10.95 %
Q32_B_Part_16 ---> Google Cloud SQL  ---> 12.64 %
Q32_B_Part_17 ---> Google Cloud Firestore  ---> 6.26 %
Q32_B_Part_18 ---> Google Cloud BigTable  ---> 7.89 %
Q32_B_Part_19 ---> Google Cloud Spanner  ---> 4.82 %
Q32_B_Part_20 ---> None ---> 7.21 %
Q32_B_OTHER ---> Other ---> 0.98 %


### Q32B - En que productos de big data esperas familiarizarte en los próximos 2 años
    Q32_B_Part_1 ---> MySQL  ---> 23.62 %
    Q32_B_Part_2 ---> PostgreSQL  ---> 12.66 %
    Q32_B_Part_3 ---> SQLite  ---> 11.69 %
    Q32_B_Part_4 ---> Oracle Database  ---> 8.87 %
    Q32_B_Part_5 ---> MongoDB  ---> 16.51 %
    Q32_B_Part_6 ---> Snowflake  ---> 4.18 %
    Q32_B_Part_7 ---> IBM Db2  ---> 4.6 %
    Q32_B_Part_8 ---> Microsoft SQL Server  ---> 12.1 %
    Q32_B_Part_9 ---> Microsoft Azure SQL Database  ---> 10.32 %
    Q32_B_Part_10 ---> Microsoft Azure Cosmos DB  ---> 5.97 %
    Q32_B_Part_11 ---> Amazon Redshift  ---> 4.66 %
    Q32_B_Part_12 ---> Amazon Aurora  ---> 4.55 %
    Q32_B_Part_13 ---> Amazon DynamoDB  ---> 5.75 %
    Q32_B_Part_14 ---> Amazon RDS  ---> 8.64 %
    Q32_B_Part_15 ---> Google Cloud BigQuery  ---> 10.95 %
    Q32_B_Part_16 ---> Google Cloud SQL  ---> 12.64 %
    Q32_B_Part_17 ---> Google Cloud Firestore  ---> 6.26 %
    Q32_B_Part_18 ---> Google Cloud BigTable  ---> 7.89 %
    Q32_B_Part_19 ---> Google Cloud Spanner  ---> 4.82 %
    Q32_B_Part_20 ---> None ---> 7.21 %
    Q32_B_OTHER ---> Other ---> 0.98 %


### Q34B - En que herramientas de business intelligence esperas familiarizarte en los próximos 2 años

---

In [11]:
nulos('Q34_B_Part_1','Q34_B_OTHER')

Q34_B_Part_1 ---> Microsoft Power BI ---> 18.95 %
Q34_B_Part_2 ---> Amazon QuickSight ---> 4.88 %
Q34_B_Part_3 ---> Google Data Studio ---> 15.75 %
Q34_B_Part_4 ---> Looker ---> 2.15 %
Q34_B_Part_5 ---> Tableau ---> 20.87 %
Q34_B_Part_6 ---> Salesforce ---> 4.87 %
Q34_B_Part_7 ---> Tableau CRM ---> 6.54 %
Q34_B_Part_8 ---> Qlik ---> 3.01 %
Q34_B_Part_9 ---> Domo ---> 1.06 %
Q34_B_Part_10 ---> TIBCO Spotfire ---> 1.03 %
Q34_B_Part_11 ---> Alteryx  ---> 2.03 %
Q34_B_Part_12 ---> Sisense  ---> 1.21 %
Q34_B_Part_13 ---> SAP Analytics Cloud  ---> 5.17 %
Q34_B_Part_14 ---> Microsoft Azure Synapse  ---> 6.53 %
Q34_B_Part_15 ---> Thoughtspot  ---> 1.01 %
Q34_B_Part_16 ---> None ---> 11.77 %
Q34_B_OTHER ---> Other ---> 1.12 %


### Q34B - En que herramientas de business intelligence esperas familiarizarte en los próximos 2 años
    Q34_B_Part_1 ---> Microsoft Power BI ---> 18.95 %
    Q34_B_Part_2 ---> Amazon QuickSight ---> 4.88 %
    Q34_B_Part_3 ---> Google Data Studio ---> 15.75 %
    Q34_B_Part_4 ---> Looker ---> 2.15 %
    Q34_B_Part_5 ---> Tableau ---> 20.87 %
    Q34_B_Part_6 ---> Salesforce ---> 4.87 %
    Q34_B_Part_7 ---> Tableau CRM ---> 6.54 %
    Q34_B_Part_8 ---> Qlik ---> 3.01 %
    Q34_B_Part_9 ---> Domo ---> 1.06 %
    Q34_B_Part_10 ---> TIBCO Spotfire ---> 1.03 %
    Q34_B_Part_11 ---> Alteryx  ---> 2.03 %
    Q34_B_Part_12 ---> Sisense  ---> 1.21 %
    Q34_B_Part_13 ---> SAP Analytics Cloud  ---> 5.17 %
    Q34_B_Part_14 ---> Microsoft Azure Synapse  ---> 6.53 %
    Q34_B_Part_15 ---> Thoughtspot  ---> 1.01 %
    Q34_B_Part_16 ---> None ---> 11.77 %
    Q34_B_OTHER ---> Other ---> 1.12 %

### Q36B - En que categorías de herramientas de machine learning esperas familiarizarte en los próximos 2 años

---

In [12]:
nulos('Q36_B_Part_1','Q36_B_OTHER')

Q36_B_Part_1 ---> Automated data augmentation (e.g. imgaug, albumentations) ---> 13.46 %
Q36_B_Part_2 ---> Automated feature engineering/selection (e.g. tpot, boruta_py) ---> 14.05 %
Q36_B_Part_3 ---> Automated model selection (e.g. auto-sklearn, xcessiv) ---> 18.33 %
Q36_B_Part_4 ---> Automated model architecture searches (e.g. darts, enas) ---> 8.4 %
Q36_B_Part_5 ---> Automated hyperparameter tuning (e.g. hyperopt, ray.tune, Vizier) ---> 12.82 %
Q36_B_Part_6 ---> Automation of full ML pipelines (e.g. Google Cloud AutoML, H2O Driverless AI) ---> 18.3 %
Q36_B_Part_7 ---> None ---> 12.4 %
Q36_B_OTHER ---> Other ---> 1.24 %


### Q36B - En que categorías de herramientas de machine learning esperas familiarizarte en los próximos 2 años
    Q36_B_Part_1 ---> Automated data augmentation (e.g. imgaug, albumentations) ---> 13.46 %
    Q36_B_Part_2 ---> Automated feature engineering/selection (e.g. tpot, boruta_py) ---> 14.05 %
    Q36_B_Part_3 ---> Automated model selection (e.g. auto-sklearn, xcessiv) ---> 18.33 %
    Q36_B_Part_4 ---> Automated model architecture searches (e.g. darts, enas) ---> 8.4 %
    Q36_B_Part_5 ---> Automated hyperparameter tuning (e.g. hyperopt, ray.tune, Vizier) ---> 12.82 %
    Q36_B_Part_6 ---> Automation of full ML pipelines (e.g. Google Cloud AutoML, H2O Driverless AI) ---> 18.3 %
    Q36_B_Part_7 ---> None ---> 12.4 %
    Q36_B_OTHER ---> Other ---> 1.24 %

### Q37B - En que herramientas específicas de machine learning esperas familiarizarte en los próximos 2 años

---

In [14]:
nulos('Q37_B_Part_1','Q37_B_OTHER')

Q37_B_Part_1 --->  Google Cloud AutoML  ---> 18.55 %
Q37_B_Part_2 --->  H2O Driverless AI   ---> 5.34 %
Q37_B_Part_3 --->  Databricks AutoML  ---> 6.35 %
Q37_B_Part_4 --->  DataRobot AutoML  ---> 5.95 %
Q37_B_Part_5 --->   Amazon Sagemaker Autopilot  ---> 10.33 %
Q37_B_Part_6 --->   Azure Automated Machine Learning  ---> 12.14 %
Q37_B_Part_7 ---> None ---> 5.38 %
Q37_B_OTHER ---> Other ---> 1.08 %


### Q37B - En que herramientas específicas de machine learning esperas familiarizarte en los próximos 2 años
    Q37_B_Part_1 --->  Google Cloud AutoML  ---> 18.55 %
    Q37_B_Part_2 --->  H2O Driverless AI   ---> 5.34 %
    Q37_B_Part_3 --->  Databricks AutoML  ---> 6.35 %
    Q37_B_Part_4 --->  DataRobot AutoML  ---> 5.95 %
    Q37_B_Part_5 --->   Amazon Sagemaker Autopilot  ---> 10.33 %
    Q37_B_Part_6 --->   Azure Automated Machine Learning  ---> 12.14 %
    Q37_B_Part_7 ---> None ---> 5.38 %
    Q37_B_OTHER ---> Other ---> 1.08 %

### Q38B - En que herramientas de gestion de machine learning esperas familiarizarte en los próximos 2 años

---

In [16]:
nulos('Q38_B_Part_1','Q38_B_OTHER')

Q38_B_Part_1 --->  Neptune.ai  ---> 4.91 %
Q38_B_Part_2 --->  Weights & Biases  ---> 6.09 %
Q38_B_Part_3 --->  Comet.ml  ---> 2.44 %
Q38_B_Part_4 --->  Sacred + Omniboard  ---> 2.28 %
Q38_B_Part_5 --->  TensorBoard  ---> 16.32 %
Q38_B_Part_6 --->  Guild.ai  ---> 2.81 %
Q38_B_Part_7 --->  Polyaxon  ---> 2.84 %
Q38_B_Part_8 --->  ClearML  ---> 3.93 %
Q38_B_Part_9 --->  Domino Model Monitor  ---> 2.56 %
Q38_B_Part_10 --->  MLflow  ---> 10.58 %
Q38_B_Part_11 ---> None ---> 17.49 %
Q38_B_OTHER ---> Other ---> 1.45 %


### Q38B - En que herramientas de gestion de machine learning esperas familiarizarte en los próximos 2 años
    Q38_B_Part_1 --->  Neptune.ai  ---> 4.91 %
    Q38_B_Part_2 --->  Weights & Biases  ---> 6.09 %
    Q38_B_Part_3 --->  Comet.ml  ---> 2.44 %
    Q38_B_Part_4 --->  Sacred + Omniboard  ---> 2.28 %
    Q38_B_Part_5 --->  TensorBoard  ---> 16.32 %
    Q38_B_Part_6 --->  Guild.ai  ---> 2.81 %
    Q38_B_Part_7 --->  Polyaxon  ---> 2.84 %
    Q38_B_Part_8 --->  ClearML  ---> 3.93 %
    Q38_B_Part_9 --->  Domino Model Monitor  ---> 2.56 %
    Q38_B_Part_10 --->  MLflow  ---> 10.58 %
    Q38_B_Part_11 ---> None ---> 17.49 %
    Q38_B_OTHER ---> Other ---> 1.45 %